packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -109,11 +109,31 @@ async function invokeUserFunction(functionArnEnv: string, payload: any) {
   const jsonPayload = parseJsonPayload(resp.Payload);
   if (resp.FunctionError) {
     log('user function threw an error:', resp.FunctionError);
+
     const errorMessage = jsonPayload.errorMessage || 'error';
-    const trace = jsonPayload.trace ? '\nRemote function error: ' + jsonPayload.trace.join('\n') : '';
 
-    const e = new Error(errorMessage);
-    e.stack += trace;
+    // parse function name from arn
+    // arn:${Partition}:lambda:${Region}:${Account}:function:${FunctionName}
+    const arn = functionArn.split(':');
+    const functionName = arn[arn.length - 1];
+
+    // append a reference to the log group.
+    const message = [
+      errorMessage,
+      '',
+      `Logs: /aws/lambda/${functionName}`,  // cloudwatch log group
+      '',
+    ].join('\n');
+
+    const e = new Error(message);
+
+    // the output that goes to CFN is what's in `stack`, not the error message.
+    // if we have a remote trace, construct a nice message with log group information
+    if (jsonPayload.trace) {
+      // skip first trace line because it's the message
+      e.stack = [ message, ...jsonPayload.trace.slice(1) ].join('\n');
```

Comment:
This used to be `+=`, is the change intentional? This won't include the local trace

Reply:
Yes it’s intentional. The local trace is not interesting as it’s always the custom resource provider framework code and users won’t know what to do about it anyway

Reply:
Fair enough